pkg/descheduler/descheduler.go

```diff
@@ -358,74 +317,82 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
-						EvictSystemCriticalPods: evictSystemCriticalPods,
-						IgnorePvcPods:           ignorePvcPods,
-						EvictFailedBarePods:     evictBarePods,
-						NodeFit:                 nodeFit,
-						PriorityThreshold: &api.PriorityThreshold{
-							Value: &thresholdPriority,
-						},
+		// Build plugins
+		for _, profile := range deschedulerPolicy.Profiles {
+			pc := getPluginConfig(defaultevictor.PluginName, profile.PluginConfigs)
+			if pc == nil {
+				klog.ErrorS(fmt.Errorf("unable to get plugin config"), "skipping plugin", "plugin", defaultevictor.PluginName)
+				continue
+			}
+			evictorFilter, err := defaultevictor.New(
+				pc.Args,
+				&handleImpl{
+					clientSet:                 rs.Client,
+					getPodsAssignedToNodeFunc: getPodsAssignedToNode,
+					sharedInformerFactory:     sharedInformerFactory,
+				},
+			)
+			if err != nil {
+				klog.ErrorS(fmt.Errorf("unable to construct a plugin"), "skipping plugin", "plugin", defaultevictor.PluginName)
+				continue
+			}
+			handle := &handleImpl{
+				clientSet:                 rs.Client,
+				getPodsAssignedToNodeFunc: getPodsAssignedToNode,
+				sharedInformerFactory:     sharedInformerFactory,
+				evictor: &evictorImpl{
+					podEvictor:    podEvictor,
+					evictorFilter: evictorFilter.(framework.EvictorPlugin),
+				},
+			}
+			// Assuming only a list of enabled extension points.
+			// Later, when a default list of plugins and their extension points is established,
+			// compute the list of enabled extension points as (DefaultEnabled + Enabled - Disabled)
+			for _, plugin := range append(profile.Plugins.Deschedule.Enabled, profile.Plugins.Balance.Enabled...) {
+				pc := getPluginConfig(plugin, profile.PluginConfigs)
+				if pc == nil {
+					klog.ErrorS(fmt.Errorf("unable to get plugin config"), "skipping plugin", "plugin", plugin)
+					continue
+				}
+				pg := pluginsMap[plugin](pc.Args, handle)
+				if pg != nil {
+					switch v := pg.(type) {
+					case framework.DeschedulePlugin:
+						enabledDeschedulePlugins = append(enabledDeschedulePlugins, v)
+					case framework.BalancePlugin:
+						enabledBalancePlugins = append(enabledBalancePlugins, v)
+					default:
+						klog.ErrorS(fmt.Errorf("unknown plugin extension point"), "skipping plugin", "plugin", plugin)
 					}
+				}
+			}
+		}
 
-					evictorFilter, _ := defaultevictor.New(
-						defaultevictorArgs,
-						&handleImpl{
-							clientSet:                 rs.Client,
-							getPodsAssignedToNodeFunc: getPodsAssignedToNode,
-							sharedInformerFactory:     sharedInformerFactory,
-						},
-					)
-
-					handle := &handleImpl{
-						clientSet:                 rs.Client,
-						getPodsAssignedToNodeFunc: getPodsAssignedToNode,
-						sharedInformerFactory:     sharedInformerFactory,
-						evictor: &evictorImpl{
-							podEvictor:    podEvictor,
-							evictorFilter: evictorFilter.(framework.EvictorPlugin),
-						},
-					}
+		// Execute extension points
+		for _, pg := range enabledDeschedulePlugins {
```

Comment:
splitting it to `enabledDeschedulePlugins` and `enabledBalancePlugins` causes it to lose the sorting capability, right? Meaning that I can never have a Balance plugin run before a Deschedule plugin

Reply:
Yes. This is a breaking change. The Balance extension points are always expected to be run after all Deschedule extension points.